app/services/foreman/plugin/role_lock.rb

```diff
@@ -19,6 +19,7 @@ def register_role(name, permissions, role_registry, description = '')
       def process_role(name, permissions, description = '')
         role = Role.find_by :name => name
         if role
+          role.lock!
```

Comment:
Can you help me understand when this lock would be released?

Reply:
It would be released at the end of the surrounding transaction.